performancecopilot/pcp
# disk.all metrics fix, qa

Changes committed to git@github.com:kmcdonell/pcp.git 20200522

Ken McDonell (7):
      qa/check-flakey: tweak some pretty printing for tests seq #s < 1000
      Linux PMDA: fix bad metadata for disk.all.*(discard|flush)* metrics
      qa/badarchives/bozo-vm-disk: new archive with incorrect disk.all metadata
      qa/1024.out: remade after fixes to disk.all metadata
      qa/1198: (new) check pmlogrewrite rules after recent Linux PMDA changes
      qa/920: fix another #! botch
      src/pmlogger/src/logue.c: fix compilation warning

 qa/1024.out                          |   84 +++---
 qa/1198                              |   47 +++
 qa/1198.out                          |  468 +++++++++++++++++++++++++++++++++++
 qa/920                               |    2 
 qa/badarchives/bozo-vm-disk.0        |binary
 qa/badarchives/bozo-vm-disk.index    |binary
 qa/badarchives/bozo-vm-disk.meta     |binary
 qa/check-flakey                      |    7 
 qa/group                             |    1 
 src/pmdas/linux/GNUmakefile          |    2 
 src/pmdas/linux/disk_all_fixups.conf |    9 
 src/pmdas/linux/pmda.c               |   14 -
 src/pmlogger/src/logue.c             |    2 
 13 files changed, 584 insertions(+), 52 deletions(-)

Details ...

commit c0a7fa14ad684123b16d9142c04e877f126a44d0
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Fri May 22 09:46:49 2020 +1000

    src/pmlogger/src/logue.c: fix compilation warning

commit 2801bcc9acac64831551bb6b8549b14bed2faa32
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Fri May 22 09:46:26 2020 +1000

    qa/920: fix another #! botch

commit a07867a85d69e763ae182275f8a48ebdd7712f8f
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Fri May 22 09:45:12 2020 +1000

    qa/1198: (new) check pmlogrewrite rules after recent Linux PMDA changes
    
    After the Linux PMDA was updated with the indom metadata
    corrected for the disk.all.*(discard|flush)* metrics.

commit 09c76e404dd38d4f4636036b300cf693a863569f
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Fri May 22 09:43:20 2020 +1000

    qa/1024.out: remade after fixes to disk.all metadata
    
    Remade after the Linux PMDA was updated with the indom metadata
    corrected for the disk.all.*(discard|flush)* metrics.

commit 75454f2f6c15ca6c2d2b0f698811c93038cb1037
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Fri May 22 09:40:56 2020 +1000

    qa/badarchives/bozo-vm-disk: new archive with incorrect disk.all metadata
    
    This archive was created before the metadata for the
    disk.all.*(discrd|flush)* metrics was corrected.
    
    To be used in the (new) qa/1198.

commit 4101311a477e902d6c3e5fbf7a141eb35b3107da
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Fri May 22 09:38:30 2020 +1000

    Linux PMDA: fix bad metadata for disk.all.*(discard|flush)* metrics
    
    They incorrectly had the an indom of DISK_INDOM instead of INDOM_NULL.
    
    pmlogrewrite rules included as well.
    
    Issue found by finally digging to understand why QA 1024 was failing on
    bozo-vm (the problem is in the linux PMDA, not the QA test).

commit 7ed13af6456a0158045feefba45b21a16a833728
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Fri May 22 09:35:35 2020 +1000

    qa/check-flakey: tweak some pretty printing for tests seq #s < 1000
